jsonrpc2.go

```diff
@@ -644,8 +644,13 @@ func (m *anyMessage) UnmarshalJSON(data []byte) error {
 		if len(msgs) == 0 {
 			return errors.New("jsonrpc2: invalid empty batch")
 		}
-		for _, msg := range msgs {
-			if err := checkType(&msg); err != nil {
+		for i := range msgs {
+			if err := checkType(&msg{
```

Comment:
In a second thought... why don't we just use, do scopelint complain about this:
```suggestion
			if err := checkType(msgs[i]); err != nil {
```

Reply:
@unknwon checkType whants pointer to ```msg```

Reply:
What about `&msgs[i]`, does scopelint complain this?

Reply:
@unknwon yep, last one works, ```&msgs[i]```

Reply:
but I am actually not sure that this is a good idea, pointer to a value inside slice that can move its values inside memory, can this actually be an issue in this case?

Reply:
If `checkType` is read-only, I think that's fine. But yeah, what's in the PR feels safer.